superset/jinja_context.py

```diff
@@ -602,3 +604,35 @@ def get_template_processor(
     else:
         template_processor = NoOpTemplateProcessor
     return template_processor(database=database, table=table, query=query, **kwargs)
+
+
+def dataset_macro(
+    dataset_id: int,
+    include_metrics: bool = False,
+    groupby: Optional[List[str]] = None,
```

Comment:
actually I was thinking, should we replace `groupby` with `columns` here? It feels it kinda makes more sense, as you should be able to pick a subset of columns from the dataset even when you don't have metrics with a GROUP BY